src/discord/rest_api/utils.rs

```diff
@@ -43,29 +39,28 @@ pub async fn download_image(
     Ok(())
 }
 
-pub fn init_data(token: &String) -> Result<Vec<ApiResponse>, std::io::Error> {
-    let token_arc = Arc::new(token.to_owned());
-
+pub fn init_data() -> Result<Vec<ApiResponse>, io::Error> {
     let (tx, rx) = oneshot::channel();
 
     runtime().spawn(async move {
-        let mut headers = HashMap::new();
-        headers.insert("Authorization", token_arc.to_string());
-
-        let channels = ApiEndpoints::GetChannels(None)
-            .call(headers.clone())
-            .await
-            .unwrap();
-
+        let channels = match ApiEndpoints::GetChannels(None).call(None).await {
```

Comment:
I believe you can remove the Ok statement by just using `.unwrap_or_else`

Reply:
gives me an error of tx being moved